src/rust/ide/lib/view/src/graph_editor/component/type_coloring.rs

```diff
@@ -0,0 +1,55 @@
+//! This module contains functionality that allows ports and edges to be colored according
+//! to their type information.
+
+use crate::prelude::*;
+
+use crate::graph_editor::SharedHashMap;
+use crate::graph_editor::Type;
+
+use ensogl::data::color;
+use std::collections::hash_map::DefaultHasher;
+use std::hash::Hasher;
+
+
+/// Color that should be used if no type information is available.
+pub const DEFAULT_TYPE_COLOR : color::Lcha = color::Lcha::new(0.5, 0.0, 0.0, 1.0);
+
+
+
+// ================================
+// === Type to Color Conversion ===
+// ================================
+
+fn type_to_hash(type_information:Type) -> u64 {
+    let mut hasher = DefaultHasher::new();
+    type_information.0.hash(&mut hasher);
+    hasher.finish()
+}
```

Comment:
why `Type` jsut does not implement hash insteaD?

Reply:
Moreover, what is the relation of this function and the section name it is placed in ?